### DISCUSSION TITLE
how to randomize blocks?

### DISCUSSION THREAD

**Discussion by roeegilron** [jspsych/jsPsych#849] [Q&A] [answered]
Hi All, 

Total newbie here so excuse ignorance :-) 

I am trying to randomize blocks of trials rather than the trials themselves. Is it possible to do this ?

I am building up from a trial:

```js     
	// right button:
        let blueRightButtonPrep = {
            type: 'html-button-response',
            stimulus: '<span class="red fixation-dimensions"></span>',
            choices: [''],
            trial_duration: PREP_DURATION,
            margin_vertical: -120,
        margin_horizontal: 0,
        button_html: '<button class="blue-rightbtn target-dimensions">%choice%</button>',
        on_finish: function (data) {
            times += Date.now().toString() + ", right target PREP end\n";
            console.log('blueTopButton prep end');
        },
        on_load: function (data) {
            times += Date.now().toString() + ", right target PREP start\n";
            console.log('blueTopButton prep start');
        }
    };
```

To a block (for brevity not including other `html-button-response` variables - `fixation`, `blueRightButtonMove`): 

```js
    //right button block
    let right_button = {
            timeline: [fixation, blueRightButtonPrep, blueRightButtonMove],
            randomize_order: false,
            repetitions: 1
    };
```

To a series of blocks. Ideally what I want to do is just randomize the order of the blocks below: `right_button`, `top_button`, `buttom_button`, `left_button` and repeat these a number of times. But the code snippet below always repeats these blocks in the same order. 

```js
   let blk1 = {
            timeline: [right_button, top_button, buttom_button, left_button],
            randomize_order: true,
            repetitions: 1
    };
```

I think I am missing something fundamental.... 

Thank You! 

p.s. you can see how the experiment flows here: 
https://roeegilron.github.io/sampleMovementTask/index.html

**Comment by jodeleeuw** [comment, marked as answer]
Hi @roeegilron ,

The `randomize_order` parameter for a timeline applies only to [timeline variables](https://www.jspsych.org/overview/timeline/#timeline-variables). If you want to randomize the order of trials/blocks on a timeline then you'll need to do something like:

```js
let blk1 = {
            timeline: jsPsych.randomization.shuffle([right_button, top_button, bottom_button, left_button]),
            repetitions: 1
};
```

Note that this will shuffle the array *once* and the same order would apply if you increased the `repetitions` parameter.
